### DISCUSSION TITLE
Why use dotenvy over doteny_macro?

### DISCUSSION THREAD

**Discussion by Dosx001** [allan2/dotenvy#90] [Q&A] [answered]
Unlike doteny, doteny_macro works both with cargo run and cargo build.

**Comment by allan2** [comment]
Can you provide more details? I'm not sure I understand.

**Comment by Dosx001** [comment]
@allan2 
```rust
fn main() {
    dotenvy::dotenv().ok();
    let secret = dotenvy::var("SECRET").expect("SECRET must be set");
    println!("{}", secret);
}
```
```bash
./target/debug/env
hello
```

```bash
cd ..
./env/target/debug/env
thread 'main' panicked at src/main.rs:3:41:
SECRET must be set: EnvVar(NotPresent)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

If I replaced dotenvy with dotenvy_macro, I can run the program anywhere and package it for another system.

**Reply by allan2** [reply, marked as answer]
There is a discussion on this [here](https://github.com/tauri-apps/tauri/discussions/6936) on the Tauri repo.

dotenvy_macro loads at compile-time whilst dotenv loads at runtime. If you make a minimal reproducible example, perhaps in a new repo, I can take a closer look.
